core/domain/user_domain.py

```diff
@@ -833,39 +872,52 @@ def remove_learnt_topic_id(self, learnt_topic_id):
         self.learnt_topic_ids.remove(learnt_topic_id)
 
 
+class LearnerGoalsDict(TypedDict):
+    """Dictionary representing the LearnerGoals object."""
+
+    user_id: str
+    topic_ids_to_learn: List[str]
+    topic_ids_to_master: List[str]
+
+
 class LearnerGoals:
     """Domain object for the learner goals model."""
 
     def __init__(
-            self, user_id, topic_ids_to_learn,
-            topic_ids_to_master):
+        self,
+        user_id: str,
+        topic_ids_to_learn: List[str],
+        topic_ids_to_master: List[str]
+    ) -> None:
         self.id = user_id
         self.topic_ids_to_learn = topic_ids_to_learn
         self.topic_ids_to_master = topic_ids_to_master
 
-    def add_topic_id_to_learn(self, topic_id):
+    def add_topic_id_to_learn(self, topic_id: str) -> None:
         """Adds the topic id to 'topic IDs to learn' list.
 
         Args:
             topic_id: str. The topic id to be inserted to the learn list.
         """
         self.topic_ids_to_learn.append(topic_id)
 
-    def remove_topic_id_from_learn(self, topic_id):
+    def remove_topic_id_from_learn(self, topic_id: str) -> None:
         """Removes the topic id from the 'topic IDs to learn' list.
 
         topic_id: str. The id of the topic to be removed.
         """
         self.topic_ids_to_learn.remove(topic_id)
 
-    def to_dict(self):
+    def to_dict(self) -> LearnerGoalsDict:
         """Return dictionary representation of LearnerGoals.
 
         Returns:
             dict. A dictionary containing the LearnerGoals class information
             in a dictionary form.
         """
-        return {
+        # Here, we're not returning user_id key of UserSettingsDict
+        # which causes MyPy to throw error. thus we add an ignore here.
+        return { # type: ignore[typeddict-item]
```

Comment:
Can we create a separate `TypedDict` then?

Reply:
Yeah, Done!